### PR TITLE
update docstrings to match new expr repr

### PR DESCRIPTION
**Pull request by gforsyth**
I haven't gone through everything, but the largest chunk of repr docstrings are in `ibis/expr/` and I've updated all of those to reflect the new expr repr.  I also ran across a repr that needed `ops.TimeAdd` so I've added that to the mapping of binop formatting options

### CODE REVIEW

File: ibis/expr/types/core.py [hunk 1]
```diff
@@ -353,11 +340,7 @@ def _binop(
     >>> import ibis.expr.operations as ops
     >>> expr = _binop(ops.TimeAdd, ibis.time("01:00"), ibis.interval(hours=1))
     >>> expr
-    time = TimeAdd
-      left:
-        value: time = datetime.time(1, 0)
-      right:
-        value: interval<int8>(unit='h') = 1
+    datetime.time(1, 0) + 1
```
Comment (cpcloud) [inline ibis/expr/types/core.py:343]: We should definitely repr the `1` here more intelligently. I'll make an issue.

**Review by cpcloud — APPROVED** [review]
Excellent.

**Comment by gforsyth** [comment]
I'm going to move the `NORMALIZE_WHITESPACE` and friends into the `pyproject.toml` and the add in a CI check just to check the doctests in `ibis/expr` which covers almost all of the important ones

File: ibis/expr/types/core.py (hunk 1 above)
Comment (cpcloud) [reply ibis/expr/types/core.py:343]: Done. #3668

**Comment by gforsyth** [comment]
I opted to leave out the CI check -- I know I've felt pretty annoyed at them in the past.  But the configuration is all there if we decide to add them in, and they can be run locally without issue (for occasional checkups and cleanups)

**Pull request merged**